docs/internal/continuous-profiler.md

```diff
@@ -0,0 +1,255 @@
+# Continuous profiler
```

Comment:
Can we tell a bit more about what the continuous profiler collects or accomplishes before starting with thread sampling?

Reply:
Something like https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3219/files#r1448765491 ?